crates/proto/src/rr/dnssec/rdata/key.rs

```diff
@@ -407,21 +407,24 @@ fn test_key_usage() {
 ///
 /// ```
 #[cfg_attr(feature = "serde-config", derive(Deserialize, Serialize))]
-#[deprecated = "Deprecated by RFC3007"]
 #[derive(Debug, Default, PartialEq, Eq, Hash, Clone, Copy)]
 pub struct UpdateScope {
     /// this key is authorized to attach,
     ///   detach, and move zones by creating and deleting NS, glue A, and
     ///   zone KEY RR(s)
+    #[deprecated = "Deprecated by RFC3007"]
```

Comment:
This seems to be unrelated, and a little weird? Are `UpdateScope`s supposed to be empty after RFC 3007?

Reply:
Because of the move from `Default::default` usage it triggered a bunch of deprecated warnings. I figured this has a similar effect but doesn't eliminate the usefulness of having a complete (if deprecated) impl. The issue I've noticed in DNS is that many old servers use outdated constructs and it can be useful to support that for parsing, which is why I don't think we should just remove this.

I can revert this and try to put in some specific `#allow`s if you think that would be better.

Reply:
I have a hard time picturing how many `#[allow]`s would be needed, but it does seem preferable.

Reply:
ok, I guess it was fewer places than I thought.